### PR TITLE
Colorise keploy logo

### PR DESCRIPTION
**Pull request by AhmedLotfy02**
## Related Issue
  - Info about Issue or bug: colorise keploy logo

Closes: #1579

#### Describe the changes you've made
A clear and concise description of what you have done to successfully close your assigned issue. Any new files? or anything you feel to let us know!
-Using ANSI and color each character to be the same as the logo
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
A clear and concise description of it.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## Screenshots (if any)

|        Original         |          Updated           |
|:-----------------------:|:--------------------------:|
| **original screenshot** | <b>updated screenshot </b> |
Updated:
image

### CODE REVIEW

**Comment by AhmedLotfy02** [comment]
I have read the CLA Document and I hereby sign the CLA

File: main.go [hunk 1]
```diff
@@ -34,9 +36,51 @@ func main() {
...
 	utils.Version = version
 	if binaryToDocker := os.Getenv("BINARY_TO_DOCKER"); binaryToDocker != "true" {
-		fmt.Println(logo, " ")
+		// ANSI escape code to reset color
+		const reset = "\033[0m"
+
+		// Print each line of the logo with a different color from the gradient
+		lines := strings.Split(logo, "\n")
+		fmt.Println(len(lines))
+		for i, line := range lines {
```
Comment (shivamsouravjha) [inline main.go:55]: seems to be unnecessary

File: go.mod [hunk 2]
```diff
@@ -26,11 +26,15 @@ require (
...
 	golang.org/x/crypto v0.11.0 // indirect
-	golang.org/x/sys v0.10.0
+	golang.org/x/sys v0.12.0
 	google.golang.org/protobuf v1.30.0 // indirect
 )
 
-require github.com/yudai/golcs v0.0.0-20170316035057-ecda9a501e82 // indirect
+require (
+	github.com/aymanbagabas/go-osc52/v2 v2.0.1 // indirect
+	github.com/charmbracelet/lipgloss v0.9.1 // indirect
```
Comment (shivamsouravjha) [inline go.mod:35]: installed but not used.

General suggestion would be to use more trusted libraries as many repos might have security vulnerability

File: main.go (hunk 1 above)
Comment (AhmedLotfy02) [reply main.go]: Regarding line 54 which is printing right?
I removed it.

File: go.mod (hunk 2 above)
Comment (AhmedLotfy02) [reply go.mod]: Thanks for advice, I forgot to remove them as I tried to show the logo as an image on the terminal using them.

Comment (shivamsouravjha) [reply go.mod]: you can try to make the color more closer,with more accurate gradients.

Comment (AhmedLotfy02) [reply go.mod]: What about this?
image

File: go.mod [hunk 3]
```diff
@@ -26,11 +26,14 @@ require (
 	go.mongodb.org/mongo-driver v1.11.6
 	go.uber.org/zap v1.24.0
 	golang.org/x/crypto v0.11.0 // indirect
-	golang.org/x/sys v0.10.0
+	golang.org/x/sys v0.12.0
 	google.golang.org/protobuf v1.30.0 // indirect
 )
```
Comment (shivamsouravjha) [inline go.mod:32]: any reason for this to be re-introduced?

File: main.go [hunk 4]
```diff
@@ -34,9 +36,67 @@ func main() {
...
+		lines := strings.Split(logo, "\n")
+		for i, line := range lines {
+			for j, char := range line {
+
+				var color string
+				if i == 7 {
+					if j > 42 {
+						color = gradientColors[0]
+					} else {
+						color = gradientColors[3]
```
Comment (shivamsouravjha) [inline main.go:61]: would suggest some more in-detail

image
Kind of sample cli images can be built via the ANSCII codes. So something that resembles more with Keploy's logo would be possible too,

File: go.mod (hunk 2 above)
Comment (shivamsouravjha) [reply go.mod]: something more resembling to keploy's logo would be possible too. the code snippet I shared was a sample of how to proceed. More refactoring can achieve the desired results.

File: go.mod (hunk 3 above)
Comment (AhmedLotfy02) [reply go.mod:32]: No I will return it to it's default

**Comment by AhmedLotfy02** [comment]
@shivamsouravjha I moved the printing logic into root.go and call it inside main the function accepts a boolean var to decide whether print colored or not

**Comment by gouravkrosx** [comment]
@AhmedLotfy02 Please update your PR according to the refactored changes.

**Comment by AhmedLotfy02** [comment]
@gouravkrosx Done.

**Comment by re-Tick** [comment]
@shivamsouravjha what is the status for the PR.

**Comment by shivamsouravjha** [comment]
@AhmedLotfy02 can you print this logo with disableANSI? flag?Incase if the flag is disabled we shouldn't colorise the logo

**Comment by shivamsouravjha** [comment]
Hey @AhmedLotfy02 I was able to run the changes locally but here are my take-aways
* PrintKeployLogo function works opposite to what it should, reason being if disableAnsi is true it would print coloured logo.
* `keploy --disableANSI=false` even this prints coloured logo.


Would suggest a recheck around the logic of printing logo.

**Comment by gouravkrosx** [comment]
@AhmedLotfy02, Can you please make a fresh PR for this.

**Pull request closed without merging**